### ISSUE TITLE
interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod

### ISSUE THREAD

**Issue by dxsup** [KindlingProject/kindling#445]
**Describe the bug**

When the watch is closed and misses the delete event, the agent panicked with the following errors:
```go
E0210 13:13:12.053218    1995 runtime.go:79] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x103d7b760), concrete:(*runtime._type)(0x103e4cdc0), asserted:(*runtime._type)(0x103fa2be0), missingMethod:""} (interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod)
goroutine 91 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x103dd48a0?, 0x140008f2ff0})
	/Users/dxsup/Codes/go/pkg/mod/k8s.io/kubernetes/staging/src/k8s.io/apimachinery@v0.0.0-20220503133649-4ce5a8954017/pkg/util/runtime/runtime.go:75 +0x7c
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x104af36c0?})
	/Users/dxsup/Codes/go/pkg/mod/k8s.io/kubernetes/staging/src/k8s.io/apimachinery@v0.0.0-20220503133649-4ce5a8954017/pkg/util/runtime/runtime.go:49 +0x7c
panic({0x103dd48a0, 0x140008f2ff0})
	/usr/local/go/src/runtime/panic.go:838 +0x204
github.com/Kindling-project/kindling/collector/pkg/metadata/kubernetes.PodDelete({0x103e4cdc0?, 0x140003a8780?})
	/Users/dxsup/Codes/kindling/collector/pkg/metadata/kubernetes/pod_watch.go:369 +0x8b0
g
```

**What did you expect to see?**

Not panic.

**Comment by dxsup** [comment]
This is the [designed behavior](https://github.com/kubernetes/client-go/blob/a406c934270567da417090023915212c73a54825/tools/cache/controller.go#L208-L211) of the `onDelete` function. We have to check if the interface received is `DeletedFinalStateUnknown` before proceeding further.

**Comment by dxsup** [comment]
Closed via #456
